File: demos/jans-chip/ios/Jans-Chip/Models/DCRequest.swift
//
//  DCRequest.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 12.10.2023.
//

import Foundation

public struct DCRequest: Codable {
    
    private enum CodingKeys: String, CodingKey {
        case issuer
        case redirectUris = "redirect_uris"
        case scope
        case responseTypes = "response_types"
        case postLogoutRedirectUris = "post_logout_redirect_uris"
        case grantTypes = "grant_types"
        case applicationType = "application_type"
        case clientName = "client_name"
        case tokenEndpointAuthMethod = "token_endpoint_auth_method"
        case evidence
        case jwks
    }
    
    var issuer: String
    var redirectUris: [String]
    var scope: [String]
    var responseTypes: [String]
    var postLogoutRedirectUris: [String]
    var grantTypes: [String]
    var applicationType: String
    var clientName: String
    var tokenEndpointAuthMethod: String
    var evidence: String = ""
    var jwks: String = ""
    
    var asParameters: [String: Any] {
        [
            "issuer": issuer,
            "redirect_uris": redirectUris,
            "scope": scope,
            "response_types": responseTypes,
            "post_logout_redirect_uris": postLogoutRedirectUris,
            "grant_types": grantTypes,
            "application_type": applicationType,
            "client_name": clientName,
            "token_endpoint_auth_method": tokenEndpointAuthMethod,
            "evidence": evidence,
            "jwks": jwks
        ]
//        "issuer=\(issuer)&redirect_uris=\(redirectUris)&scope=\(scope)&response_types=\(responseTypes)&post_logout_redirect_uris=\(postLogoutRedirectUris)&grant_types=\(grantTypes)&application_type=\(applicationType)&client_name=\(clientName)&token_endpoint_auth_method=\(tokenEndpointAuthMethod)&evidence=\(evidence)&jwks=\(jwks)"
    }
}

File: demos/jans-chip/kmp/fido2/src/commonMain/kotlin/com/example/fido2/model/DCRequest.kt
package com.example.fido2.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class DCRequest (
    @SerialName("issuer")
    var issuer: String? = null,

    @SerialName("redirect_uris")
    var redirectUris: List<String?>,

    @SerialName("scope")
    var scope: String? = null,

    @SerialName("response_types")
    var responseTypes: List<String?>,

    @SerialName("post_logout_redirect_uris")
    var postLogoutRedirectUris: List<String?>,

    @SerialName("grant_types")
    var grantTypes: List<String?>,

    @SerialName("application_type")
    var applicationType: String? = null,

    @SerialName("client_name")
    var clientName: String? = null,

    @SerialName("token_endpoint_auth_method")
    var tokenEndpointAuthMethod: String? = null,

    @SerialName("evidence")
    var evidence: String? = null,

    @SerialName("jwks")
    var jwks: String? = null,
)